### PR TITLE
[Testerina] Handle empty data providers

### LINKED ISSUE #32575: Tests with data providers fails when provided with zero items
Tests with data providers fails when provided with zero items

Eg:
```ballerina
import ballerina/test;

@test:Config {
    dataProvider: provider
}
function test(int i) returns error? {
}

function provider() returns map<[int]> {
    return {};
}
```

It just fails with `error: there are test failures` without even a good explanation.

### PR DESCRIPTION
**Pull request by Dilhasha**
## Purpose
Handle empty data providers
Fixes #32575

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/BTestRunner.java [hunk 1]
```diff
@@ -564,24 +564,35 @@ private void executeFunction(Test test, TestSuite suite, String packageName, Cla
...
+                        List<String> keyValues = getKeyValues((BMap) valueSets);
+                        Class<?>[] argTypes = extractArgumentTypes((BMap) valueSets);
+                        List<Object[]> argList = extractArguments((BMap) valueSets);
+                        int i = 0;
+                        for (Object[] arg : argList) {
+                            invokeDataDrivenTest(suite, test.getTestName(), escapeSpecialCharacters(keyValues.get(i)),
+                                    classLoader, scheduler, shouldSkip, packageName, arg, argTypes,
+                                    failedOrSkippedTests);
+                            i++;
+                        }
```
Comment (NipunaRanasinghe) [inline misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/BTestRunner.java:580]: We can avoid the temp variable `i` by using an optimised indexed for loop instead. WDYT?

File: misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/BTestRunner.java [hunk 2]
```diff
@@ -564,24 +564,35 @@ private void executeFunction(Test test, TestSuite suite, String packageName, Cla
...
+                    } else {
+                        // Handle array data sets
+                        Class<?>[] argTypes = extractArgumentTypes((BArray) valueSets);
+                        List<Object[]> argList = extractArguments((BArray) valueSets);
+                        int i = 0;
+                        for (Object[] arg : argList) {
+                            invokeDataDrivenTest(suite, test.getTestName(), String.valueOf(i), classLoader, scheduler,
+                                    shouldSkip, packageName, arg, argTypes, failedOrSkippedTests);
+                            i++;
+                        }
```
Comment (NipunaRanasinghe) [inline misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/BTestRunner.java:595]: https://github.com/ballerina-platform/ballerina-lang/pull/32625/files#r704170958

Comment (Dilhasha) [reply misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/BTestRunner.java]: Fixed with 92b98e5037942465b58076ce069597e5f12ae983

File: misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/BTestRunner.java (hunk 1 above)
Comment (Dilhasha) [reply misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/BTestRunner.java]: Fixed with 92b98e5037942465b58076ce069597e5f12ae983

File: misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/BTestRunner.java [hunk 3]
```diff
@@ -564,24 +564,31 @@ private void executeFunction(Test test, TestSuite suite, String packageName, Cla
...
+                    } else {
+                        // Handle array data sets
+                        Class<?>[] argTypes = extractArgumentTypes((BArray) valueSets);
+                        List<Object[]> argList = extractArguments((BArray) valueSets);
+                        for (int i = 0, argListSize = argList.size(); i < argListSize; i++) {
+                            invokeDataDrivenTest(suite, test.getTestName(), String.valueOf(i), classLoader, scheduler,
+                                    shouldSkip, packageName, argList.get(i), argTypes, failedOrSkippedTests);
+                        }
                     }
                 } else if (valueSets instanceof BError || valueSets instanceof Error ||
```
Comment (pramodya1994) [inline misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/BTestRunner.java:593]: Can't we remove this condition? `valueSets instanceof BError`
I think it'scovered by this `valueSets instanceof Exception`

Comment (Dilhasha) [reply misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/BTestRunner.java]: Fixed with https://github.com/Dilhasha/plugin-vscode.git

**Review by NipunaRanasinghe — COMMENTED** [review]
LGTM!

**Review by pramodya1994 — APPROVED** [review]
LGTM

**Pull request merged**